apis/v1alpha2/validation/httproute.go

```diff
@@ -212,5 +212,23 @@ func validateHTTPPathModifier(modifier gatewayv1a2.HTTPPathModifier, path *field
 	if modifier.ReplacePrefixMatch == nil && modifier.Type == gatewayv1a2.PrefixMatchHTTPPathModifier {
 		errs = append(errs, field.Invalid(path, modifier.ReplacePrefixMatch, "must not be nil if the HTTPRouteFilter.Type is ReplacePrefixMatch"))
 	}
+
+	if len(errs) == 0 && modifier.ReplacePrefixMatch != nil {
+		if !hasExactlyOnePrefixMatch(matches) {
```

Comment:
Why not add this error to the list of errors?

Reply:
It reads to me like this adds the error to the list of errors in the case that there's not exactly one prefix match that's not replaced, which seems right. @robscott, maybe some comments here would help make it clearer?

Reply:
Good call, this was confusing. I was actually using `len(errs) == 0` as a shortcut for a check on `modifier.Type == gatewayv1a2.PrefixMatchHTTPPathModifier` but that ended up being more confusing than it was worth. Updated to just use the latter check since it's much easier to understand.